marlowe-dashboard-client/src/Contract/View.purs

```diff
@@ -30,57 +30,84 @@ import Halogen.Extra (lifeCycleEvent)
 import Halogen.HTML.Events.Extra (onClick_, onValueInput_)
 import Halogen.HTML.Properties (InputType(..), enabled, href, placeholder, ref, target, type_, value)
-import Humanize (formatDate, formatTime, humanizeDuration, humanizeInterval, humanizeValue)
+import Humanize (contractIcon, formatDate, formatTime, humanizeDuration, humanizeInterval, humanizeValue)
 import Marlowe.Execution.Lenses (_semanticState, _mNextTimeout)
 import Marlowe.Execution.State (expandBalances, getActionParticipant)
 import Marlowe.Execution.Types (NamedAction(..))
 import Marlowe.Extended (contractTypeName)
+import Marlowe.Extended.Metadata (_contractType)
 import Marlowe.PAB (transactionFee)
 import Marlowe.Semantics (Accounts, Assets, Bound(..), ChoiceId(..), Input(..), Party(..), Slot, SlotInterval, Token, TransactionInput(..), getEncompassBound)
 import Marlowe.Slot (secondsDiff, slotToDateTime)
-import Material.Icons (Icon(..), icon)
+import Material.Icons (Icon(..)) as Icon
+import Material.Icons (Icon, icon)
 import WalletData.State (adaToken, getAda)
 
--- I'm moving this view here so that we can easily call Contract.Actions from inside it. I'm not
--- actually calling any Contract.Actions from here yet, but that's a TODO for the next PR...
-contractInnerBox :: forall p. Slot -> State -> HTML p Action
-contractInnerBox currentSlot state =
+contractCard :: forall p. Slot -> State -> HTML p Action
+contractCard currentSlot state =
   let
-    nickname = state ^. _nickname
-
-    mMarloweParams = state ^. _mMarloweParams
-
-    stepNumber = currentStep state + 1
+    currentStepNumber = (currentStep state) + 1
 
-    mNextTimeout = state ^. (_executionState <<< _mNextTimeout)
+    nickname = state ^. _nickname
 
-    timeoutStr =
-      maybe'
-        (\_ -> if isContractClosed state then "Contract closed" else "Timed out")
-        (\nextTimeout -> humanizeDuration $ secondsDiff nextTimeout currentSlot)
-        mNextTimeout
+    contractType = state ^. (_metadata <<< _contractType)
   in
-    div_
+    div
+      [ classNames [ "shadow", "bg-white", "rounded" ] ]
       [ div
-          [ classNames [ "flex-1", "px-4", "py-2", "text-lg" ] ]
-          -- TODO: make (new) nicknames editable directly from here
-          [ text if null nickname then "My new contract" else nickname ]
+          [ classNames [ "flex", "gap-2", "px-4", "py-2", "border-gray", "border-b" ] ]
+          [ div
+              [ classNames [ "flex-1" ] ]
+              [ h3
+                  [ classNames [ "flex", "gap-2", "items-center" ] ]
+                  [ contractIcon contractType
+                  , text $ contractTypeName contractType
+                  ]
+              , input
+                  [ classNames $ Css.inputNoBorder <> [ "-ml-2", "text-lg" ]
+                  , type_ InputText
+                  , value nickname
+                  , onValueInput_ SetNickname
+                  , placeholder "Please rename"
```

Comment:
is the `Please rename` something for us internal to remember we have to rename? if so, it's probably worth to add a FIXME. If it's for the end user we might want to ask Niahm for some copy

Reply:
this isn't a note for us - this is the text Russ included in his designs